src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,190 @@
+        printSeparation();
+    }
+
+    public static void updateTaskStatus(Task[] commandStorage, String line) {
+        try {
+            line = line.trim();
+            int startOfTaskIndex = line.indexOf(' ') + 1;
+            int taskIndex = Integer.parseInt(line.substring(startOfTaskIndex)) - 1;
+            commandStorage[taskIndex].markTaskAsDone();
+
+            printSeparation();
+            System.out.println("    Nice! I've marked this task as done: ");
+            System.out.format("    ");
+            printTaskDescription(commandStorage[taskIndex]);
+            printSeparation();
+        } catch (NullPointerException e) {
+            printSeparation();
+            System.out.println("    OOPS!!! The task does not exist.");
+            printSeparation();
+        } catch (ArrayIndexOutOfBoundsException e) {
+            printSeparation();
+            System.out.println("    OOPS!!! There is no such thing as task number 0.");
+            printSeparation();
+        }
+    }
+
+    private static void printTaskDescription(Task t) {
+        System.out.format("[%s][%s] %s ", t.getTaskType(), t.getStatusIcon(), t.getTaskName());
+        switch (t.getTaskType()) {
+            case "T":
+                System.out.println("");
+                break;
+            case "D":
+                System.out.println("(by: " + t.getTaskTime() + ")");
+                break;
+            case "E":
+                System.out.println("(at: " + t.getTaskTime() + ")");
+                break;
+        }
+    }
+
+    public static void addToDo(String line) {
+        try {
+            String taskName = line.substring(lengthOfTypeToDo + 1);
+            ToDo t = new ToDo(taskName);
+            addTaskToList(t);
+        } catch (StringIndexOutOfBoundsException e) {
+            printSeparation();
+            System.out.println("    OOPS!!! The description of a todo cannot be empty.");
+            printSeparation();
+        }
+    }
+
+    public static void addDeadline(String line) {
+        try {
+            int endOfTaskNameIndex = line.indexOf('/', lengthOfTypeDeadline) - 1;
+            String taskName = line.substring(lengthOfTypeDeadline+1, endOfTaskNameIndex);
+
+            int startOfTaskTimeIndex = line.indexOf("/by") + 4;
+            String taskTime = line.substring(startOfTaskTimeIndex);
+
+            Deadline d = new Deadline(taskName, taskTime);
+            addTaskToList(d);
+        } catch (StringIndexOutOfBoundsException e) {
+            printSeparation();
+            System.out.println("    OOPS!!! Both the description and time of a deadline cannot be empty.");
+            printSeparation();
+        }
+    }
+
+    public static void addEvent(String line) {
+        try {
+            int endOfTaskNameIndex = line.indexOf('/', lengthOfTypeEvent) - 1;
+            String taskName = line.substring(lengthOfTypeEvent+1, endOfTaskNameIndex);
+
+            int startOfTaskTimeIndex = line.indexOf("/at") + 4;
```

Comment:
There is a use of magic number `4` here. Perhaps there is a better way to name this number as a `CONSTANT` or have an `enum` class to handle it?